### DISCUSSION TITLE
Which variant of adapter is the fastest

### DISCUSSION THREAD

**Discussion by andrewt27** [john30/ebusd#818] [Q&A]
Which variant of the adapter is the fastest when it comes to communication with the eBUS. I have eBUS adapter 3 with a WiFi communication, which can also be connected to the USB port after disconnecting from the ESP8266. 
Will I notice a significant improvement in reading speed if I convert this adapter to a USB variant?

**Comment by cyberthom42** [comment]
I would also be interested. To expand the scope of this question a bit:
Is there a difference in speed/ reliability between the RPI version and the USB/Ethernet/Wifi version?
Currently my assumption is that running ebusd on the raspberry with the RPi HAT connected would be the fastest solution which also offers the best reliability, ist that correct?

**Comment by andrewt27** [comment]
@john30 Can you comment on the topic as the author of the device?

**Comment by Commifreak** [comment]
I think, USB would be the fastest variant (maybe the RPi variant is even better?). Followed by ethernet and last but not least: WiFi. Thats my understanding.

**Reply by andrewt27** [reply]
It's just your guess, I suspect so too, but I'd like to be sure about that.
